services/api/service.go

```diff
@@ -635,6 +750,38 @@ func (api *RelayAPI) updateProposerDuties(headSlot uint64) {
 	api.log.Infof("proposer duties updated: %s", strings.Join(_duties, ", "))
 }
 
+func (api *RelayAPI) updateOptimisticSlot(headSlot uint64) {
```

Comment:
maybe a more expressive function name?

```suggestion
func (api *RelayAPI) prepareOptimisticBuildersForSlot(headSlot uint64) {
```

Reply:
went with `prepareOptimisticSlot` since we do more than just update the builders!

Reply:
but this function is not only used for optimistic builders, it's also used to set the high/low prio status of builders in the regular flow. thus i'd argue it has very little to do with optimistic relaying at all, and should maybe be called `prepareBuildersForSlot`

Reply:
it has the critical lines

```
// Wait until there are no optimistic blocks being processed. Then we can
// safely update the slot.
api.optimisticBlocksWG.Wait()
api.optimisticSlot.Store(headSlot + 1)
```

which is the heartbeat of updating the optimistic slot. but sure. i dont mind too much. how about `prepareOptimisticBuildersForSlot` as you originally suggested.